components/brave_wallet/browser/solana_tx_manager.cc

```diff
@@ -85,19 +89,197 @@ void SolanaTxManager::AddUnapprovedTransaction(
     return;
   }
 
-  SolanaTxMeta meta(from, std::move(tx));
-  meta.set_id(TxMeta::GenerateMetaID());
-  meta.set_origin(
+  auto meta = std::make_unique<SolanaTxMeta>(from, std::move(tx));
+  meta->set_id(TxMeta::GenerateMetaID());
+  meta->set_origin(
       origin.value_or(url::Origin::Create(GURL("chrome://wallet"))));
-  meta.set_created_time(base::Time::Now());
-  meta.set_status(mojom::TransactionStatus::Unapproved);
-  meta.set_chain_id(chain_id);
-  if (!tx_state_manager_->AddOrUpdateTx(meta)) {
+  meta->set_created_time(base::Time::Now());
+  meta->set_status(mojom::TransactionStatus::Unapproved);
+  meta->set_chain_id(chain_id);
+
+  // If the transaction is partially signed, we can't modify the instructions
+  // to add a priority fee.
+  if (meta->tx()->IsPartialSigned()) {
+    if (!tx_state_manager_->AddOrUpdateTx(*meta)) {
+      std::move(callback).Run(
+          false, "", l10n_util::GetStringUTF8(IDS_WALLET_INTERNAL_ERROR));
+      return;
+    }
+    std::move(callback).Run(true, meta->id(), "");
+    return;
+  }
+
+  // Fetch the compute units used for priority fee estimation
+  // by using the simulateTransaction API.
```

Comment:
I think we can remove this comment now that we have better comment in GetSolanaTxFeeEstimationForMeta.